pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -258,6 +262,50 @@ public synchronized String submitTask(List<PinotTaskConfig> pinotTaskConfigs, St
     return taskStates;
   }
 
+  /**
+   * This method helps compute the percentage done for a task (that may have many partitions or sub-tasks).
+   * @param parentTaskName (e.g. "Task_TestTask_1624403781879")
+   * @return a pair of integers, first one being the total number partitions and the second being number of partitions
+   * still running or yet to run.
+   */
+  public synchronized TaskCount getTaskCount(String parentTaskName) {
+    JobContext jobContext = _taskDriver.getJobContext(getHelixJobName(parentTaskName));
+    TaskCount taskCount = new TaskCount();
+    for (int partition : jobContext.getPartitionSet()) {
+      taskCount.addToTotal(1);
```

Comment:
`taskCount.addToTotal(jobContext.getPartitionSet().size());` out of the for loop?

Reply:
Done